bookserver/routers/assessment.py

```diff
@@ -143,3 +166,355 @@ async def get_history(request: Request, request_data: HistoryRequest):
     ]
 
     return make_json_response(detail=res)
+
+
+# Used by :ref:`compareAnswers`
+@router.get("/getaggregateresults")
+async def getaggregateresults(request: Request, div_id: str, course_name: str):
+    question = div_id
+
+    if not request.state.user:
+        return make_json_response(
+            status=status.HTTP_401_UNAUTHORIZED,
+            detail=dict(answerDict={}, misc={}, emess="You must be logged in"),
+        )
+
+    if course_name in (
+        "thinkcspy",
+        "pythonds",
+        "fopp",
+        "csawesome",
+        "apcsareview",
+        "StudentCSP",
+    ):
+        start_date = datetime.datetime.utcnow() - datetime.timedelta(days=90)
+    else:
+        course = await fetch_course(course_name)
+        start_date = course.term_start_date
+
+    result = await count_useinfo_for(question, course_name, start_date)
+
+    tdata = {}
+    tot = 0.0
+    for row in result:
+        tdata[row[0]] = row[1]
+        tot += row[1]
+
+    tot = float(tot)
+    rdata: Dict[str, float] = {}
+    miscdata = {}
+    correct = ""
+    if tot > 0:
+        for key in tdata:
+            all_a = key.split(":")
+            try:
+                answer = all_a[1]
+                if "correct" in key:
+                    correct = answer
+                count = float(tdata[key])
+                if answer in rdata:
+                    count += rdata[answer] / 100.0 * tot
+                pct = round(count / tot * 100.0)
+
+                if answer != "undefined" and answer != "":
+                    rdata[answer] = pct
+            except Exception as e:
+                rslogger.error("Bad data for %s data is %s -- %s" % (question, key, e))
+
+    miscdata["correct"] = correct
+    miscdata["course"] = course
+
+    returnDict = dict(answerDict=rdata, misc=miscdata)
+
+    # if instructor:
+    # There is little value to doing this now when the instructor
+    # Dashboard provides more and better detail
+    #     resultList = _getStudentResults(question)
+    #     returnDict["reslist"] = resultList
+
+    return make_json_response(detail=returnDict)
+
+
+@router.get("/getpollresults")
+async def getpollresults(request: Request, course: str, div_id: str):
+
+    # fetch summary of poll answers
+    result = await fetch_poll_summary(div_id, course)
+
+    opt_counts = {}
+
+    for row in result:
+        rslogger.debug(row)
+        val = int(row[0])
+        opt_counts[val] = row[1]
+
+    opt_num = max(opt_counts.keys()) if opt_counts else 0
+    for i in range(opt_num):
+        if i not in opt_counts:
+            opt_counts[i] = 0
+    # opt_list holds the option numbers from smallest to largest
+    # count_list[i] holds the count of responses that chose option i
+    opt_list = sorted(opt_counts.keys())
+    count_list = []
+    for i in opt_list:
+        count_list.append(opt_counts[i])
+
+    total = sum(opt_counts.values())
+    user_res = None
+    if request.state.user:
```

Comment:
This seems a bit more complicated than necessary -- I would use:
``` Python
    if request.state.user:
        user_res = await fetch_last_poll_response(
            request.state.user.username, course, div_id
        )
        my_vote = int(user_res)
    else:
        my_vote = -1
```

Reply:
These are all useful comments but I struggle with taking code that has worked for years and redoing NOW just for to get past your code review.  I have lots of other tasks on my plate.

Reply:
I understand -- time is a precious resource. I'm approving this PR and will leave all this to your discretion.